io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/request/OIDCRequestBuilder.java

```diff
@@ -81,16 +83,32 @@ public OIDCRequestBuilder(OIDCAgentConfig oidcAgentConfig) {
         Scope authScope = oidcAgentConfig.getScope();
         URI callBackURI = oidcAgentConfig.getCallbackUrl();
         URI authorizationEndpoint = oidcAgentConfig.getAuthorizeEndpoint();
-        State state = generateStateParameter();
+        String stateParam = oidcAgentConfig.getState();
+        State state;
+        if (StringUtils.isBlank(stateParam)) {
+            state = generateStateParameter();
+        } else {
+            state = new State(stateParam);
+        }
```

Comment:
shall we extract this out to a separate private method like `resolveState(oidcAgentConfig)` and call it here?

Reply:
fixed in e7e548c